### ISSUE TITLE
add ESM, support for module and browser

### ISSUE THREAD

**Issue by kopax-polyconseil** [TehShrike/deepmerge#250]
Hello and thanks for sharing,

As I can see when working on [this](https://github.com/callstack/react-theme-provider/issues/141) matter, you don't have any `module` field in package.json for serving ESM, the most recent version of JS modules.

You also don't have a `browser` field for serving the umd.

Do you have any plan to add those or is this module obselete?

**Comment by RebeccaStevens** [comment]
For version 4 there are no plans to add this (see #194). If version 5 every comes out (see #241) it should have this.

In the meantime you can use https://github.com/RebeccaStevens/deepmerge-ts

**Comment by jorenbroekema** [comment]
https://www.npmjs.com/package/@bundled-es-modules/deepmerge feel free to use ESM bundled fork of this project.

I felt I had no choice but to do this, since the author seems reluctant to add ESM-first support, and deepmerge-ts is almost 4 times as large as original deepmerge.. Of course, you can also commit yourself to a compile-step like @rollup/plugin-commonjs or webpack/esbuild equivalent but for projects that are otherwise compile-step-free, that's not a great option either.

I will deprecate my fork as soon as the author of this package publishes an ESM version.

**Comment by agrohs** [comment]
Is there a fork in github w/ the actual changes in your npm package @jorenbroekema??

**Comment by jorenbroekema** [comment]
> Is there a fork in github w/ the actual changes in your npm package @jorenbroekema??

yes https://github.com/bundled-es-modules/deepmerge , sorry must have forgotten to put this info in the pkg json

all I do is run esbuild (bundler) over the exported deepmerge pkg and format as ESM, then dual publish both ESM / CJS.
